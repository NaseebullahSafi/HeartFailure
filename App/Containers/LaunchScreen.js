import React, { Component, PureComponent } from 'react'
import {View, TouchableHighlight, StyleSheet, Text, Animated} from 'react-native'
import BLE from './BLE';
import Ionicons from 'react-native-vector-icons/Feather';
import { TabViewAnimated, TabBar } from 'react-native-tab-view';
import type { NavigationState } from 'react-native-tab-view/types';
import SimplePage from './SimplePage';
/*
* ======= Firebase Initialisation ======
* */
import firebase from 'react-native-firebase';
import CustomModal from "../Components/CustomModal";

// firebase config
const firebaseConfig = {
  apiKey: "AIzaSyDFdv9jmcfZzzRs2OHu_j3ZXV0gkz5HniQ",
  authDomain: "heartfailure-bad0c.firebaseapp.com",
  databaseURL: "https://heartfailure-bad0c.firebaseio.com",
  projectId: "heartfailure-bad0c",
  storageBucket: "heartfailure-bad0c.appspot.com",
};

type Route = {
  key: string,
  icon: string,
};

type State = NavigationState<Route>;

export default class LaunchScreen extends Component<*, State> {

  static title = 'Icon only top bar';
  static appbarElevation = 0;

  constructor(props) {
    super(props);
    this.state = {
      index: 0,
      routes: [{
        key: '1', icon: 'activity'
      }, {
        key: '2', icon: 'airplay'
      }],
      selectedItem: 'About',
      isPressed: false,
      currentUser: null,
      modalVisible: false,
      loading: false,
      userType: "",
      Patients: null,
      Users: null,
      Doctors: null
    };
    this.usersRef             = firebase.app().database().ref();
    this.listenForPersons     = this.listenForPersons.bind(this);
  }

  componentWillMount() {
    this.listenForPersons(this.usersRef);
  };

  listenForPersons = (personsRef) => {
    personsRef.on('value', (snap) => {
      this.setState({ Users : snap.val().Users, Patients : snap.val().Patients, Doctors : snap.val().Doctors }, () => {
        this.currentUser().catch(e => console.log(e));
      });
    });
  };

  _handleIndexChange = index => { this.setState({ index }) };


  updateIndex = () => { this.setState({ index: this.state.index === 0 ? 1 : 0 }) };


  _renderIcon = ({ route }) => { return <Ionicons name={route.icon} size={24} color="#bccad0" />; };

  openModel = () => { this.child.toggleModal();};

  _renderHeader = props => {

    return (
      <View style={{elevation: 2, backgroundColor:'white'}}>
        <View style={styles.headerContainer}>
          <TouchableHighlight
            onPress={() => {
              this.props.navigation.navigate('DrawerToggle')
            }}
            activeOpacity={1.0} underlayColor="rgba(253,138,94,0)">
            <Ionicons style={{padding: 8}} name="menu" size={22} color="#bccad0"/>
          </TouchableHighlight>
          <View style={{position: 'relative'}}>
            <TouchableHighlight activeOpacity={1} underlayColor="rgba(253,138,94,0)" onPress={() => this.openModel()}>
              <Ionicons style={[styles.msgIcon, this.state.isPressed ? styles.testing : {}]} name="message-square" size={22} color="#bccad0"/>
            </TouchableHighlight>
            <View style={styles.notificationDot} />
          </View>
        </View>
        <TabBar
          {...props}
          indicatorStyle={styles.indicator}
          renderIcon={this._renderIcon}
          style={styles.tabbar}
        />
      </View>
    );
    /*<CustomModal onRef={ref => this.child = ref}/>*/
  };

  currentUser = async () => {
    var user = firebase.app().auth().currentUser;
    if (user) {
      this.setState({
        userType: this.state.Patients[user._user.uid] ? (
          "Patient"
        ): (
          this.state.Doctors[user._user.uid] ? "Doctor" : ""
        ),
        currentUser: user
      })
    }
  };

  _renderScene = ({ route }) => {
    return (
      <SimplePage
        state       = {this.state}
        style       = {{ backgroundColor: 'white' }}
        userType    = {this.state.userType}
        updateIndex = {this.updateIndex.bind(this)}
        currentUser = {this.state.currentUser}
        Patients    = {this.state.Patients}
        Doctors     = {this.state.Doctors}
      />
    );
  };

  onMenuItemSelected = item => {
    this.setState({
      selectedItem: item,
    });
  };

  _onHideUnderlay = () => {
    this.setState({ isPressed: false });
  };

  _onShowUnderlay = () => {
    this.setState({ isPressed: true });

  };

  render() {
    return (
      <TabViewAnimated
        style={[styles.container, this.props.style]}
        navigationState={this.state}
        renderScene={this._renderScene}
        renderHeader={this._renderHeader}
        onIndexChange={this._handleIndexChange}
      />
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  tabbar: {
    backgroundColor: 'white',
    paddingTop: 10,
    elevation: 0
  },
  notificationDot: {
    position: 'absolute',
    right: 7,
    top: 5,
    width: 12,
    height: 12,
    borderColor: 'white',
    borderWidth: 3,
    borderRadius: 100/2,
    backgroundColor: '#E67D8F'
  },
  indicator: {
    backgroundColor: 'rgba(152, 168, 171, 0)'
  },
  headerContainer: {
    height: 60,
    flexDirection: 'row',
    justifyContent: 'space-between',
    backgroundColor: 'white',
    alignItems: 'center',
    width: '100%',
    paddingLeft: 30,
    paddingRight: 30,
  },
  headerText: {
    fontSize: 20,
    fontWeight: 'bold',
    color: '#8395ab',
  },
  button: {
    position: 'absolute',
    top: 10,
    padding: 10,
  },
  msgIcon: {
    padding: 8,
  },
  testing: {
    alignItems: 'center',
    borderColor: '#fff',
    justifyContent: 'center',
    padding: 8,
    backgroundColor: '#f9f9fa',
    borderRadius: 100 / 2,
  }
});
