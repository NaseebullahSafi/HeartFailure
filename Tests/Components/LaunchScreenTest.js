// need to import React itself
import React from 'react'

// everything fro react native
import 'react-native'

// required for mocking modules
import Enzyme,{ shallow } from 'enzyme'

// needed to pass components to the renderer
import renderer from 'react-test-renderer'

// adapter is needed for this react version
import Adapter from 'enzyme-adapter-react-16';

// we cant use firebase in jest, so we have mocked it
import firebase from '../../firebase'

// the component that will be tested
import LaunchScreen from "../../App/Containers/LaunchScreen";

// instantiate an adapter
Enzyme.configure({ adapter: new Adapter() });

/**
 * Test the entire component to see if it renders properly
 */
describe('Testing <LaunchScreen /> component', () => {

  // create an image of the component, and match it with the snapshot
  it('renders as expected', () => {
    const tree = renderer.create(
      <LaunchScreen />
    ).toJSON();
    expect(tree).toMatchSnapshot()
  });

  /**
   * Set of methods from the component is tested below
   */
  it('retrieveInfo() renders accordingly', () => {
    const wrapper = shallow(<LaunchScreen />);
    const inst = wrapper.instance().retrieveInfo(firebase.database().ref(`/Users/`));
    expect(inst)
  });

  it('_handleIndexChange() renders accordingly', () => {
    const wrapper = shallow(<LaunchScreen />);
    const inst = wrapper.instance()._handleIndexChange(1);
    expect(inst)
  });

  it('updateIndex() renders accordingly', () => {
    const wrapper = shallow(<LaunchScreen />);
    const inst = wrapper.instance().updateIndex("Doctor");
    expect(inst)
  });

  it('toggleTitle() renders accordingly', () => {
    const wrapper = shallow(<LaunchScreen />);
    const inst = wrapper.instance().toggleTitle("My Patients");
    expect(inst)
  });

  it('updateUserView() renders accordingly', () => {
    const wrapper = shallow(<LaunchScreen />);
    const inst = wrapper.instance().updateUserView({});
    expect(inst)
  });

  it('_openRightSideMenu() renders accordingly', () => {
    const navigation = { navigate: jest.fn() };
    const wrapper = shallow(<LaunchScreen navigation={navigation} />);
    const inst = wrapper.instance()._openRightSideMenu();
    expect(inst)
  });

  it('_renderHeader() renders accordingly', () => {
    const wrapper = shallow(<LaunchScreen />);
    const inst = wrapper.instance()._renderHeader();
    expect(inst)
  });

});
