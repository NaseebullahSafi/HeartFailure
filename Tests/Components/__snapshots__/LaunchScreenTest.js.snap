// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <LaunchScreen /> component renders as expected 1`] = `
<View
  loaded={
    Array [
      0,
    ]
  }
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
        "overflow": "hidden",
      },
      Object {
        "flex": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "space-between",
          "paddingLeft": 30,
          "paddingRight": 30,
          "width": "100%",
        },
        Object {
          "backgroundColor": "white",
          "elevation": 0.3,
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#bccad0",
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "Feather",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#bccad0",
          "fontSize": 22,
        }
      }
    />
    <View
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={null}
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#bccad0",
                "fontSize": 22,
              },
              Object {
                "padding": 8,
              },
              Object {
                "fontFamily": "Feather",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#E67D8F",
            "borderColor": "white",
            "borderRadius": 50,
            "borderWidth": 3,
            "height": 12,
            "position": "absolute",
            "right": 7,
            "top": 5,
            "width": 12,
          }
        }
      />
    </View>
  </View>
  <RCTScrollView
    alwaysBounceHorizontal={false}
    automaticallyAdjustContentInsets={false}
    bounces={false}
    contentContainerStyle={
      Object {
        "flex": 1,
      }
    }
    contentOffset={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    directionalLockEnabled={true}
    horizontal={true}
    keyboardDismissMode="on-drag"
    keyboardShouldPersistTaps="always"
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    overScrollMode="never"
    pagingEnabled={true}
    scrollEnabled={undefined}
    scrollEventThrottle={1}
    scrollsToTop={false}
    showsHorizontalScrollIndicator={false}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "flex": 1,
            "overflow": "hidden",
          }
        }
        testID={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "stretch",
              "backgroundColor": "#fafafa",
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#fafafa",
                "flex": 1,
                "flexDirection": "column",
                "position": "relative",
              }
            }
          >
            <RCTScrollView
              contentContainerStyle={
                Object {
                  "flexGrow": 1,
                }
              }
              showsVerticalScrollIndicator={false}
            >
              <View>
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#fafafa",
                        "flex": 1,
                        "flexDirection": "column",
                        "position": "relative",
                      },
                      Object {
                        "alignItems": "center",
                        "flex": 1,
                        "flexDirection": "row",
                        "justifyContent": "center",
                        "padding": 30,
                      },
                    ]
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#8F9CAE",
                          "fontSize": 33,
                        },
                        undefined,
                        Object {
                          "fontFamily": "Feather",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }
                  >
                    
                  </Text>
                </View>
              </View>
            </RCTScrollView>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "#6482e6",
                  "borderRadius": 300,
                  "bottom": 0,
                  "elevation": 3,
                  "flex": 1,
                  "margin": 25,
                  "opacity": 1,
                  "padding": 20,
                  "position": "absolute",
                  "right": 0,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 20,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Feather",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={null}
        testID={undefined}
      />
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "elevation": 20,
      }
    }
  >
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "white",
          "elevation": 0,
          "paddingBottom": 2,
          "paddingTop": 2,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 0.5,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 0.5,
          "zIndex": 1,
        }
      }
    >
      <View
        collapsable={undefined}
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={undefined}
          style={
            Object {
              "backgroundColor": "rgba(152, 168, 171, 0)",
              "bottom": 0,
              "height": 2,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
              "width": 0,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "overflow": "scroll",
          }
        }
      >
        <RCTScrollView
          alwaysBounceHorizontal={false}
          automaticallyAdjustContentInsets={false}
          bounces={false}
          collapsable={undefined}
          contentContainerStyle={
            Array [
              Object {
                "flexDirection": "row",
                "flexWrap": "nowrap",
              },
              Object {
                "flex": 1,
              },
            ]
          }
          contentOffset={undefined}
          horizontal={true}
          keyboardShouldPersistTaps="handled"
          onMomentumScrollBegin={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          overScrollMode="never"
          scrollEnabled={undefined}
          scrollEventThrottle={1}
          scrollsToTop={false}
          showsHorizontalScrollIndicator={false}
        >
          <View>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits="button"
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flex": 1,
                  "opacity": 1,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  collapsable={undefined}
                  style={
                    Object {
                      "alignItems": "center",
                      "flex": 1,
                      "justifyContent": "center",
                      "opacity": 1,
                      "padding": 12,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#bccad0",
                          "fontSize": 20,
                        },
                        undefined,
                        Object {
                          "fontFamily": "Feather",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }
                  >
                    
                  </Text>
                </View>
              </View>
            </View>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits="button"
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flex": 1,
                  "opacity": 1,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  collapsable={undefined}
                  style={
                    Object {
                      "alignItems": "center",
                      "flex": 1,
                      "justifyContent": "center",
                      "opacity": 0.7,
                      "padding": 12,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#bccad0",
                          "fontSize": 20,
                        },
                        undefined,
                        Object {
                          "fontFamily": "Feather",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }
                  >
                    
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </View>
</View>
`;
